lib/sly-completion.el

```diff
@@ -216,9 +216,28 @@ COMPLETIONS is a list of propertized strings."
            finally return (list formatted nil)))
 
 (defun sly-completion-annotation (completion)
-  "Grab the annotation of COMPLETION, a string, if any"
+  "Grab the annotation of COMPLETION, a string, if any."
   (get-text-property 0 'sly--annotation completion))
 
+(defun sly--completion-company-kind (completion)
+  "Convert the COMPLETION annotation to a company-kind supported symbol."
+  (let ((annot
+         (sly-completion-annotation completion)))
+    (when-let* ((comps (split-string annot " "))
+                (kind (car comps)))
+      (cond
+       ((string= kind "fn") 'function)
```

Comment:
Don't you think this is a little too much `string=`?  A constant list + `assoc` or even a `cl-loop` would do a cleaner job.   But see my other comment.

Reply:
"cleaner" is something not standardized in the coding world. I think something local and explicit even if longer can be more clean than having to look 30 lines up or down and hidden behind a few function calls.
But I don't want to debate about it. It should align into this existing code base. Insofar what you suggest is probably more aligned.

Reply:
>  hidden behind a few function calls.

`cdr` and `assoc` are 40 years old.  Nothing hidden there, everyone knows this idiom.

```
(cdr (assoc x '(("foo" . 'foo) 
                ("bar" . 'bar)) 
            #'equal))
```
Write your intent, not C :-) (though I think even C has constant expressions nowadays).

Reply:
C, what is C?
I took that "constant list" and thought you wanted that defined elsewhere in the code.
This way is fine. I'm sure I also had come up with it after some sleep. :-)

Reply:
> C, what is C?

Trick/philosophical question? :-)  C is the language where you have to write a long  `if`/`else` chain and `strcmp`

But yes, the constant list can be defined elsewhere (variable or defun) if you want, no problem with that.  It's just the repeated `string=` that struck me as odd and unidiomatic.  If you don't want `assoc` or a data list (which may or may not be compile-time constant), `pcase` is also very idiomatic IMO:

```lisp
(pcase x
  ("foo" 'foo)
  ("bar" 'bar))
```

`cl-case` can't be used here unfortunately because it uses `eql`

Reply:
It's `pcase`.

Reply:
I've read somewhere that Richard Stallman wants to replace `pcase` with some `cond` extended variant and now there are both in Emacs codebase.

Reply:
Yeah, that stupid shit.  Use `pcase`, please.  We need to be compatible to  older Emacsen anyway.

Reply:
Yep. Not sure if you've seen. I've updated the PR.